paperqa/agents/search.py

```diff
@@ -326,6 +333,13 @@ async def _add_document() -> None:
             )
             raise
 
+    async def checkpoint(self) -> None:
```

Comment:
Can you call `checkpoint` in `delete_document` too

Reply:
Yes, but for that `delete_document` will have to be async and not be a static method, is that ok?

Reply:
Oh yeah `delete_document` doesn't need to be a `staticmethod`